### PR TITLE
#4663(1) Patient Vaccine card

### LINKED ISSUE #4663: Vaccination card (read only) UI
List all doses for all vaccine courses in the immunisation program. Order by age range, then by suggested date:

Suggested date should be the date patient reaches the age range milestone, or the date once min interval since last dose has past (whichever is later). Suggested date should only be provided for the next dose of each vaccine course. This should be a value calculated on backend and just displayed in UI.

(ignore the progress bar for now)
Screenshot 2024-08-22 at 5 32 06 PM

(Joining on vaccinations to show the given doses can come later)

### PR DESCRIPTION
**Pull request by CarlosNZ**
Fixes #4663 -- partially

# 👩🏻‍💻 What does this PR do?

Just the bare minimum here, including hard-coded hook result. It's just a standard table at this point, next step is to implement styling and cell formatting.

## 💌 Any notes for the reviewer?

Lots of To-Dos, probably wouldn't want to merge this until subsequent PRs included. Do we want to start a feature branch @lache-melvin ?

# 🧪 Testing

- [ ] Clicking on Patient Program from Patient's "Vaccination Cards" tab takes you to Vaccination Card detail view (with hard-coded data currently)
- [ ] URL contains Patient ID and Program Enrolment ID
- [ ] Breadcrumb updated to show Vaccination Card and the name of the Program

# TO-DO
- [x] Create GraphQL node to return correct data
- [x] Update `usePatientVaccineCard` hook to fetch actual data (using current API pattern)
- [ ] Vax card table formatting
  - [x] Age nicely expressed in years/months (details TBC -- suggest using just months until 2 years, then years/months after that)
  - [x] Table row background colour based on status
  - [x] "Chip" tokens (new cell) for "Status" field
  - [x] General "Date" cell (I thought we would have had one of these, but there's just an "Expiry Date" cell currently
  - [ ] Add "Batch" and "Facility" fields
  - [ ] (If time and possible with current components): Figure out how to vertically merge "Age" cell across multiple dose rows (as per design)

### CODE REVIEW

File: client/packages/system/src/Patient/VaccinationCards/DetailView/DetailView.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      // Cell: DateCell, TO-DO
+    },
+  ]);
+
+  useEffect(() => {
+    if (data)
+      setCustomBreadcrumbs({
+        1: data?.vaccineCardItems?.patient?.name ?? '',
+        2: t('label.vaccination-card'),
+        3: data?.vaccineCardItems?.programName,
```
Comment (lache-melvin) [inline client/packages/system/src/Patient/VaccinationCards/DetailView/DetailView.tsx:68]: Clicking on the vaccination card part of the breadcrumb:
Screenshot 2024-09-11 at 11 50 58 AM

Takes you here:
Screenshot 2024-09-11 at 11 50 53 AM

I think either need to 
- disable the link on that crumb (had a quick look, looks like we have a `disabled` field in the breadcrumb state, but haven't implemented a nice way to set that yet, could make a refactor issue? maybe can do something like what was done in [ContactTrace](https://github.com/msupply-foundation/open-msupply/blob/4663-patient-vax-card/client/packages/system/src/ContactTrace/DetailView/DetailView.tsx#L115-L149))
- or add a redirect back to patient detail view if they end up on the `{patientId}/vaccine-card` route

**Review by lache-melvin — APPROVED** [review]
Looking good Carl! A few comments, but happy for them to be addressed in future PRs. 

>  probably wouldn't want to merge this until subsequent PRs included. Do we want to start a feature branch

I think given this is all new pages/components, and nothing is breaking, it's all good to go into develop. We've tried not to have feature branches where possible to reduce the conflicts when merging 👍

File: client/packages/system/src/Patient/VaccinationCards/DetailView/DetailView.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+export const VaccinationCardComponent: FC = () => {
+  const t = useTranslation('dispensary');
+  const { patientId = '', programEnrolmentId = '' } = useParams();
+  const { setCustomBreadcrumbs } = useBreadcrumbs();
+
+  const {
+    query: { data, isLoading },
+  } = usePatientVaccineCard(patientId, programEnrolmentId);
+
+  const tableData = buildTableData(data);
```
[the remark this answers is not in the document]
Comment (CarlosNZ) [reply client/packages/system/src/Patient/VaccinationCards/DetailView/DetailView.tsx:31]: Should be sorted by your PR and my subsequent one(s).

**Review by lache-melvin — APPROVED** [review]
Looks awesome, thanks Carl! 🚢

**Pull request merged**